### ISSUE TITLE
MacOs - Toggle Window in System Tray does not always work

### ISSUE THREAD

**Issue by smimish** [keepassxreboot/keepassxc#2823]
## Expected Behavior
When I click "Toggle Window" from the system tray icon, I should always get a focus to the KeepassXC app.

## Current Behavior
When I have multiple desktops in MacOS, and when KeepassXC is open in one of them and I am working in another, and when I click on a system tray icon -> "Toggle Window", I do not get switched to the Keepass app.

## Steps to Reproduce
1. Open KeepassXC app in MacOS
2. Create a new Desktop, switch to it
3. Click on a system tray icon -> Toggle window
nothing will happen

## Debug Info
KeePassXC - 2.4.0-beta2

Operating system: macOS

**Comment by TraceyC77** [comment]
This is also happening for me on KeePassXC 2.4.3 on MacOS with 4 virtual desktops
The bug happens even without switching to another desktop

1. Wake OSX from sleep
2. Click on the KeePassXC tray icon -> Toggle Window

Nothing happens
Clicking on Toggle Window a second time brings up the KeePassXC window.

**Comment by droidmonkey** [comment]
Could it be that KeePassXC is already visible (ie not minimized) on one of the desktops so the first time you "toggle window" it actually minimizes. The second time it appears as it should. I tend to agree, however, that the app should be summoned to the currently visible desktop.

**Comment by TraceyC77** [comment]
> Could it be that KeePassXC is already visible (ie not minimized) on one of the desktops so the first time you "toggle window" it actually minimizes. The second time it appears as it should. I tend to agree, however, that the app should be summoned to the currently visible desktop.

After a little additional testing I think you're right. I usually have KeePassXC minimized to tray, undock the MacBook, and then close the lid to put it to sleep. What I did to test this in a different way:

1. Minimize KeePassXC to tray
2. Use the Apple menu to put the MacBook to sleep without undocking
3. Woke the MacBook

The KeePassXC window was up on the screen. So it appears it restored from tray on wake, but is not on the visible screens (may be on a different virtual desktop). Therefore, the first click on "toggle" is sending the window to the tray.

It seems that
- When restoring the window it should always restore to the currently active virtual desktop
- Left click should be "restore" and not bring up the right click menu (I have created another issue for that)

**Comment by droidmonkey** [comment]
This is likely no longer a problem.

**Comment by deweydb** [comment]
I still believe this behaviour is undesirable. Most system tray apps for macOS do that behave this way. The typical behaviour is that it would bring the app to the foreground. I realize that it is technically correct, that if the app is already open, but say behind some other window, and the user attempts to access it via the system tray, and they select "toggle window", technically that hides it, and yes, technically correct. However that's sort of a pointless feature that doesn't benefit the user in any way. What they really wanted to do was bring the app to the foreground. Perhaps there could be another option other than "toggle window" such as "show window" or something.

**Comment by droidmonkey** [comment]
Wouldn't that be done from the dock? If the app is open but hidden then clicking the dock icon focuses it.

**Comment by deweydb** [comment]
Yeah I get what you're saying, I guess it's just a matter of breaking a habit, all the other password managers i've used interacted directly from the status bar, you could click the icon and even search immediately in that dropdown, or use it to open the full app to do more complex actions. 
For example:

<img width="1280" height="702" alt="Image" src="https://github.com/user-attachments/assets/0e561dbe-3d57-4838-92f4-a53bdb7971c2" />

It would be amazing if the status bar component could be expanded a bit to have some of these more quick access search/lookup type features.
